### PR TITLE
Remove DGPATH

### PR DESCRIPTION
**Pull request by uphoffc**
Removes the stupid DGPATH.

Open todos:
- ~Remove f_interoperability_computeMInvJInvPhisAtSources~
- ~Fix initial loading for dg_setup in plasticity~
- ~Fix Slowness calculation (needs mass matrix) in faultouput~
- ~Test (after above todos are addressed)~
- ~Update doc~

Removes "pmethod" in parameters.par. The "average" approach is not implemented anymore, it is unnecessary, and pmethod is undocumented.

### CODE REVIEW

**Comment by Thomas-Ulrich** [comment]
youhou!!!

**Comment by krenzland** [comment]
I think SeisSol uses the new software development paradigm "code by annoyance" :)

**Comment by AliceGabriel** [comment]
oh my god.

**Comment by daisy20170101** [comment]
cool!

File: src/ResultWriter/faultoutput.f90
Comment (Thomas-Ulrich) [inline src/ResultWriter/faultoutput.f90:645]: are you sure about that?
e.g. C. Pelties dissertation, equation 2.7

Comment (uphoffc) [reply src/ResultWriter/faultoutput.f90:645]: Please don't review yet. I commented that because I removed MassMatrix_Tri and I need to replace that by something else. (One of the TODOs.)

Comment (Thomas-Ulrich) [reply src/ResultWriter/faultoutput.f90:645]: ok sorry

Comment (uphoffc) [reply src/ResultWriter/faultoutput.f90:645]: No need to apologize. I appreciate the effort :)

**Comment by git-taufiq** [comment]
Great! 👍

**Comment by uphoffc** [comment]
Ready to review now.

I tested the Northridge and TPV13 examples from last weeks training. Differences are of the order of the machine epsilon.

I would appreciate if you would test some further example before, in particular for the Vr output, point sources, and plasticity.

**Comment by Thomas-Ulrich** [comment]
Hi,
I systematically tested the Palu training setup with o4, single precision, by differentiating the fault output and got also results similar to machine precision.

File: src/SourceTerm/Manager.cpp
Comment (Thomas-Ulrich) [inline src/SourceTerm/Manager.cpp:98]: is` centre` a good name for this variable? (I guess this is not the tetra center?).

File: src/ResultWriter/faultoutput.f90
Comment (Thomas-Ulrich) [inline src/ResultWriter/faultoutput.f90:649]: are you sure about the -1?

**Review by Thomas-Ulrich — COMMENTED** [review]
Hi,

I ve looked into your code and proposed a few changes, mostly for clarity.
I think somebody from TUM might want to give a more informed review, therefore I did not approve the PR.
should pmethod be removed from the cmake ?

File: src/Solver/Interoperability.h
Comment (uphoffc) [reply src/Solver/Interoperability.h:411]: Why? The routine calls a Fortran routine in plasticity.f90 which is deleted with this PR.

File: src/SourceTerm/Manager.cpp
Comment (uphoffc) [reply src/SourceTerm/Manager.cpp:98]: You could interpret it as the centre of the (sub-)fault the point source represents.

File: src/ResultWriter/faultoutput.f90
Comment (uphoffc) [reply src/ResultWriter/faultoutput.f90:649]: Yes, in Fortran you generally start counting from 1, in C/C++ generally from 0.

**Comment by uphoffc** [comment]
"should pmethod be removed from the cmake ?"
@Thomas-Ulrich to what do you refer here?

File: src/Solver/Interoperability.h
Comment (Thomas-Ulrich) [reply src/Solver/Interoperability.h:411]: I was talking about the header below :-)
```    
* Computes mInvJInvPhisAtSources[i] = |J|^-1 * M_ii^-1 * phi_i(xi, eta, zeta),
    * where xi, eta, zeta is the point in the reference tetrahedron corresponding to x, y, z.
    *
    * @param x x coordinate
    * @param y y coordinate
    * @param z z coordinate
    * @param element Number of element in that x, y, z resides
    * @param mInvJInvPhisAtSources contains the output
````

Comment (Thomas-Ulrich) [reply src/Solver/Interoperability.h:411]: > "should pmethod be removed from the cmake ?"
> @Thomas-Ulrich to what do you refer here?

the parameter of cmake when building SeisSol:
```
PLASTICITY                       ON                                                                                                                                                                                                           
 PLASTICITY_METHOD                nb                                                                                                                                                                                                           
```

File: src/SourceTerm/Manager.cpp
Comment (Thomas-Ulrich) [reply src/SourceTerm/Manager.cpp:98]: ok I understand the logic (same name as in the NRF reader).

File: src/Solver/Interoperability.h
Comment (uphoffc) [reply src/Solver/Interoperability.h:411]: That parameter is still supported.

Comment (uphoffc) [reply src/Solver/Interoperability.h:411]: computeMInvJInvPhisAtSources is history, too. That guy moved to https://github.com/SeisSol/SeisSol/blob/carsten/rm-dgpath/src/SourceTerm/Manager.cpp#L98

Comment (Thomas-Ulrich) [reply src/Solver/Interoperability.h:411]: Ok, I thought you removed one of the PLASTIC_METHOD.
But now I realize that you were talking about something else. sorry about that.

**Comment by Thomas-Ulrich** [comment]
Ok I read that, but my point is that there is no more header in 
https://github.com/SeisSol/SeisSol/blob/carsten/rm-dgpath/src/SourceTerm/Manager.cpp#L98 
which explains that computeMInvJInvPhisAtSources does

|mInvJInvPhisAtSources[i] = |J|^-1 * M_ii^-1 * phi_i(xi, eta, zeta) |

**Review by Thomas-Ulrich — APPROVED** [review]
Given that the code works, given that no one from TUM seems to want to review, and given that I looked at the code with my limited IT skills, I give you my approve :-)

**Comment by krenzland** [comment]
I would like to review it as well today or tomorrow, if that's okay for @uphoffc.

**Review by krenzland — CHANGES_REQUESTED** [review]
Just to block merge for now - will be approved if @uphoffc disagrees as the pr is already reviewed.

File: src/Initializer/dg_setup.f90
Comment (krenzland) [inline src/Initializer/dg_setup.f90:863]: That confused me - isn't PlastMethod used anymore?

File: src/ResultWriter/faultoutput.f90
Comment (krenzland) [inline src/ResultWriter/faultoutput.f90:659]: Commented code should be removed

File: src/SourceTerm/Manager.cpp
Comment (krenzland) [reply src/SourceTerm/Manager.cpp:98]: The bigger problem is the weird British spelling here ;)

**Review by krenzland — CHANGES_REQUESTED** [review]
Looks mostly good to me! Some minor comments. By the way, a lot more variables could be declared const - I always like to do that as it can avoid some bugs.
With C++17 most parts of Functions.cpp could actually be constexpr (I think) - might be useful to remember that.

File: src/Initializer/dg_setup.f90
Comment (uphoffc) [reply src/Initializer/dg_setup.f90:863]: That parameter refers to https://github.com/SeisSol/SeisSol/blob/master/src/Solver/plasticity.f90#L286
But that routine has never been ported to C++ and is not necessary, too.

**Review by krenzland — APPROVED** [review]

**Pull request merged**
